crates/rust-project-goals-cli/src/cfp.rs

```diff
@@ -0,0 +1,533 @@
+        if let Some(this_period_match) = this_period_pattern.find(&content) {
+            // Found an existing section for this specific timeframe
+            // Find the end of this section (next section or end of file)
+            if let Some(next_section_pos) = content[this_period_match.end()..].find("\n## ") {
+                let end_pos = this_period_match.start() + next_section_pos + this_period_match.end() - this_period_match.start();
+                new_content.replace_range(this_period_match.start()..end_pos, &new_section);
+            } else {
+                // No next section, replace until the end
+                new_content.replace_range(this_period_match.start().., &new_section);
+            }
+        } else if let Some(next_period_match) = next_period_pattern.find(&content) {
+            // Found an existing "Next goal period" section
+            // Find the end of this section (next section or end of file)
+            if let Some(next_section_pos) = content[next_period_match.end()..].find("\n## ") {
+                let end_pos = next_period_match.start() + next_section_pos + next_period_match.end() - next_period_match.start();
+                new_content.replace_range(next_period_match.start()..end_pos, &new_section);
+            } else {
+                // No next section, replace until the end
+                new_content.replace_range(next_period_match.start().., &new_section);
+            }
+        } else {
+            // No existing "Next goal period" section, try to add after "Current goal period" section
+            if let Some(current_period_match) = current_period_pattern.find(&content) {
+                // Find the end of the current period section
+                if let Some(next_section_pos) = content[current_period_match.end()..].find("\n## ") {
+                    let insert_pos = current_period_match.end() + next_section_pos;
+                    new_content.insert_str(insert_pos, &new_section);
+                } else {
+                    // No next section after current period, append to the end
+                    new_content.push_str(&new_section);
+                }
+            } else {
+                // No "Current goal period" section either, add after the introduction
+                if let Some(pos) = content.find("\n## ") {
+                    new_content.insert_str(pos, &new_section);
+                } else {
+                    // Fallback: append to the end
+                    new_content.push_str(&new_section);
+                }
+            }
+        }
+        
+        new_content
+    }
+}
+
+/// Creates the directory structure and files for a new Call For Proposals (CFP) period.
+pub fn create_cfp(timeframe: &str, force: bool, dry_run: bool) -> Result<()> {
+    if dry_run {
+        println!("Dry run mode - no changes will be made");
+    }
+    // Validate the timeframe format
+    validate_timeframe(timeframe)?;
```

Comment:
By ensuring/making `timeframe` the uppercase version of the goal period, the conversion can be removed from the 3 functions that do it, and make the 1 function assuming it's already the case correct.